src/shared/utils/validators.ts

```diff
@@ -0,0 +1,40 @@
+import { ReflectClient, ReflectCursor, ReflectUserInfo } from "shared/types"
+
+export function validateReflectCursor(value: unknown): ReflectCursor {
+  if (
+    value !== null &&
+    typeof value === "object" &&
+    "x" in value &&
+    typeof value.x === "number" &&
+    "y" in value &&
+    typeof value.y === "number"
+  ) {
+    return value as ReflectCursor
+  }
+
+  return null
+}
+
+export function validateReflectUserInfo(value: unknown): ReflectUserInfo {
+  const userInfo = value as ReflectUserInfo
+
+  return {
+    name: userInfo?.name || "",
+    avatar:
+      typeof userInfo?.avatar === "string" || userInfo?.avatar === null
+        ? userInfo.avatar
+        : null,
+    stakingRealmColor: userInfo?.stakingRealmColor || "",
+    cursorTextColor: userInfo?.cursorTextColor || "",
+  }
+}
+
+export function validateReflectClientState(value: unknown): ReflectClient {
+  const clientInfo = value as ReflectClient
+
+  return {
+    id: clientInfo?.id || "",
```

Comment:
if there is no id - what does it mean? should we still try to display the cursor?